Plans grid: monthly plans copy changes

ciampo opened the pull request:
#### Changes proposed in this Pull Request

* Change copy related to the plans grid

Do not merge this PR until all translations have been completed!

## Testing instructions

### Setup
- prepare your sandbox
- in two separate terminal windows, run:
  - `yarn start` 
  - `cd apps/editing-toolkit && yarn dev --sync`

### Focused Launch

1. Assign yourself to the "treatment" group in the `focused_launch_flow_v2` A/B test (id `20106`)
2. Add an _unlaunched_ site created through `/start`to your sandbox
3. Open the block editor (e.g. by visiting `calyspo.localhost:3000/page/UNLAUNCHED_SITE_CREATED_WITH_START.wordpress.com/home`)`
4. Press "Launch" button, Focused Launch summary view should appear
5. Click on "View all plans" link
   - [x] Verify that the new copy is correct in the plans details view (table variant), according to A/C in #49390
   - [x] These copy changes are not expected to affect non-english locales for the purpose of this PR (but it's important that the previous _translated_ fallback text is used)


### Step by Step Launch

1. Add an _unlaunched_ site on a _free_ plan created through `/new` to your sandbox
2. Visit `wordpress.com/page/SITE_ID.wordpress.com/home`
3. Press "Launch" button, Step by Step Launch modal should appear.
4. Navigate to the plans step
   - [x] Verify that the new copy is correct according to A/C in #49390 (accordion variant)
   - [x] These copy changes are not expected to affect non-english locales for the purpose of this PR (but it's important that the previous _translated_ fallback text is used)

5. Navigate to the final (checkout) step
   - [x] Verify that the new copy is correct according to A/C in #49390
   - [x] These copy changes are not expected to affect non-english locales for the purpose of this PR (but it's important that the previous _translated_ fallback text is used)


### Gutenboarding

1. Visit `/new` and navigate to the plans step
   - [x] Verify that the new copy is correct in the plans details view (accordion variant), according to A/C in #49390

#### Screenshots

For translators: 

![image](https://user-images.githubusercontent.com/1083581/106131105-87ebd100-6162-11eb-9465-654e32b55f77.png)

![image](https://user-images.githubusercontent.com/1083581/106131180-a651cc80-6162-11eb-95b0-4e9da31bca35.png)

![image](https://user-images.githubusercontent.com/1083581/107206989-0caada80-6a00-11eb-890f-d435411c85d9.png)

![image](https://user-images.githubusercontent.com/1083581/107207148-3fed6980-6a00-11eb-88d0-9df2cd976f1d.png)

![image](https://user-images.githubusercontent.com/1083581/106133086-16615200-6165-11eb-8982-ebf9aba74277.png)


Fixes #49390
Relates to #47576

ciampo committed:
Toggle component: From "Pay monthly" / "Pay annually" to "Monthly" / "Annually"

ciampo committed:
Add clarity to translator comments

a8ci18n commented on apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx:
```diff
@@ -127,7 +127,11 @@ const FinalStep: React.FunctionComponent< LaunchStepProps > = ( { onNextStep, on
-						? __( 'billed annually', 'full-site-editing' )
+						? sprintf(
+								// translators: %s is the cost per year (e.g "billed as 96$ annually")
+								__( 'per month, billed as %s annually', 'full-site-editing' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'per month{{br/}}billed yearly' )` *ES Score: 7*

a8ci18n commented on packages/plans-grid/src/plans-table/plan-item.tsx:
```diff
@@ -149,12 +149,16 @@ const PlanItem: React.FunctionComponent< Props > = ( {
+											__( 'billed as %s annually', __i18n_text_domain__ ),
+											planProduct?.annualPrice
+									  )
+									: __( 'billed monthly', __i18n_text_domain__ ) ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **28** times:
`translate( 'Billed monthly' )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/feat/plans-grid-copy-changes)

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5477733

Thank you @ciampo for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5477733).

ciampo commented on packages/plans-grid/src/plans-table/plan-item.tsx:
```diff
@@ -212,7 +214,7 @@ const PlanItem: React.FunctionComponent< Props > = ( {
 											<>
 												{ isSelected ? TickIcon : '' }
 												{ isSelected
-													? sprintf( fullWidthCtaLabelSelected, name )
+													? fullWidthCtaLabelSelected
```
Removed unnecessary `sprintf` call, since `fullWidthCtaLabelSelected` doesn't have any placeholder text

StefanNieuwenhuis commented on apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-steps/final-step/index.tsx:
```diff
@@ -127,15 +127,21 @@ const FinalStep: React.FunctionComponent< LaunchStepProps > = ( { onNextStep, on
 	);
 
+	// translators: %s is the cost per year (e.g "billed as 96$ annually")
+	const planSummaryCostLabelAnnually = __(
```
I know that there's a lot of discussion going on about `i8n` and how to implement methods properly, so it's no biggie if it's not possible but can't we "merge" `planSummaryCostLabelAnnually` & `planSummarCostLabelMonthly` into 1 `const` by creating a ternary based on `planProduct.billingPeriod`?

ciampo replied:
I'm afraid we can't — the whole point of keeping these variables separate from the ternary operator is to somehow avoid an optimisation from terser that breaks `i18n` (see p1612522813165500/1612255293.049500-slack-C013QHLF28Y)

StefanNieuwenhuis replied:
Thanks for elaborating. Makes sense and I hope we find a solution in the future.

StefanNieuwenhuis commented on packages/plans-grid/src/plans-accordion-item/index.tsx:
```diff
@@ -75,6 +75,10 @@ const PlanAccordionItem: React.FunctionComponent< Props > = ( {
 	};
 
+	// translators: %s is the cost per year (e.g "billed as 96$ annually")
+	const planItemPriceLabelAnnually = __( 'billed as %s annually', __i18n_text_domain__ );
```
Same here (see my previous comment).

ciampo replied:
Same as https://github.com/Automattic/wp-calypso/pull/49397/files/2d137c3873faedb724b186c106063d507fabed6c#r571922139

StefanNieuwenhuis commented on packages/plans-grid/src/plans-table/plan-item.tsx:
```diff
@@ -105,9 +105,11 @@ const PlanItem: React.FunctionComponent< Props > = ( {
 	// translators: %s is a WordPress.com plan name (eg: Free, Personal)
 	const fullWidthCtaLabelUnselected = __( 'Select %s', __i18n_text_domain__ );
+	// translators: %s is the cost per year (e.g "billed as 96$ annually")
+	const planItemPriceLabelAnnually = __( 'per month, billed as %s annually', __i18n_text_domain__ );
```
Same here (see comment above)

ciampo replied:
Same as https://github.com/Automattic/wp-calypso/pull/49397/files/2d137c3873faedb724b186c106063d507fabed6c#r571922139

StefanNieuwenhuis reviewed (changes requested):
#### Focused Launch - Plans Grid (table variant)
The screenshots for translators don't align with the AC in #49390 and the translations in the code. Can you please update them?

<img width="750" alt="Screenshot 2021-02-08 at 10 49 03" src="https://user-images.githubusercontent.com/12104589/107204342-dae44480-69fc-11eb-8208-8b1ededebfa5.png">

<img width="766" alt="Screenshot 2021-02-08 at 10 46 36" src="https://user-images.githubusercontent.com/12104589/107204346-dc157180-69fc-11eb-998f-d16e87465716.png">

ciampo commented:
> The screenshots for translators don't align with the AC in #49390 and the translations in the code. Can you please update them?

@StefanNieuwenhuis I've updated the screenshots, and I've [replied](https://github.com/Automattic/wp-calypso/pull/49397#discussion_r571922139) to your comment —  can you please take another look?

ciampo commented:
On hold until all strings are translated

ciampo committed:
Step by Step checkout: from 'billed annually' to 'per month, billed as %s annually'

ciampo committed:
Plans grid (accordion): copy changes

- from 'billed annually' to 'billed as %s annually'
- from 'per month, billed monthly' to 'billed monthly'

ciampo committed:
Plans grid (table): from 'billed annually' to 'per month, billed as %s annually'

ciampo commented:
@Automattic/i18n There seems to be an issue with the translations on this PR — Everything should already be translated, but I did one last check before merging and found out that the copy is still appearing in english (the same issue was flagged on a different PR, https://github.com/Automattic/wp-calypso/pull/49290#issuecomment-787778243)

Can you help debugging what's the issue?

ciampo committed:
Use old translations as fallback until new translations are in

ciampo commented:
@StefanNieuwenhuis, I pushed some changes so that the UI will display the old text until the new text is fully translated. This is necessary for ETK related changes.

I updated the testing instructions reflecting the latest changes

yansern reviewed (approved):
Tested on Focused Launch & Step By Step Launch. LGTM.

ciampo committed:
Remove 2nd argument from hasTranslation call

ciampo committed:
Remove 2nd argument from hasTranslation call
